src/LineChart.ts

```diff
@@ -48,7 +49,7 @@ module n3Charts {
       scope.$watch('options+data', () => {
         // Call the update event with a copy of the options
         // to avoid infinite digest loop
-        eventMgr.trigger('update', scope.datasets, angular.copy(scope.options), attributes);
+        eventMgr.trigger('update', scope.datasets, new Utils.Options(angular.copy(scope.options)), attributes);
```

Comment:
yep, I was thinking the same. What about the dataset attribute? Should we have also some parser (that removes invisible series from the array or sth? Do we still need to pass the attributes everywhere? Maybe it's ok to just pass them to the options (where the options parser generates valid options out of the attributes).